hibernate-core/src/main/java/org/hibernate/tuple/entity/AbstractEntityTuplizer.java

```diff
@@ -569,6 +565,20 @@ protected boolean shouldGetAllProperties(Object entity) {
 			// if the attribute is not lazy (bytecode sense), we can just use the value from the instance
 			// if the attribute is lazy but has been initialized we can just use the value from the instance
 			// todo : there should be a third case here when we merge transient instances
+			if(enhancedForLazyLoading) {
+				final BytecodeLazyAttributeInterceptor interceptor = enhancementMetadata.extractLazyInterceptor(entity);
+				if (interceptor == null && lazyAttributesMetadata.isLazyAttribute( propertyName ) && getters[j].get(entity) == null) {
+					result[j] = LazyPropertyInitializer.UNFETCHED_PROPERTY;
```

Comment:
To understand why, take a look at the `isAttributeLoaded` routine used by the pre-existing code.  It returns true by default every time the interceptor is null, which is wrong, because all transient/detached objects start out with their interceptors `null` and lazy associations unpopulated.  

Due to this bad default, all of the `null` values which needed to be lazy loaded were being ignored and staying null.  Now they will be fetched, as they are supposed to be, after the interceptor has been assigned, which happens later.

Notice line 567 of the pre-existing code:
`// todo : there should be a third case here when we merge transient instances`

Yep, transient instances were not being handled well here.